### PR TITLE
fix: _add_embeddings_to_elements bug resulting in duplicated elements

### PR DESCRIPTION
**Pull request by ryannikolaidis**
Currently when the  OpenAIEmbeddingEncoder adds embeddings to Elements in `_add_embeddings_to_elements` it overwrites each Element's `to_dict` method, mistakenly resulting in each Element having identical values with the exception of the actual embedding value. This was due to the way it leverages a nested `new_to_dict` method to overwrite. Instead, this updates the original definition of Element itself to accommodate the `embeddings` field when available. This also adds a test to validate that values are not duplicated.

### CODE REVIEW

File: unstructured/documents/elements.py [hunk 1]
```diff
@@ -490,6 +490,8 @@ def to_dict(self) -> dict:
         out["element_id"] = self.id
         out["type"] = self.category
         out["text"] = self.text
```
Comment (rbiseck3) [inline unstructured/documents/elements.py:492]: Can we instead add this to the Element class:

embedding: Optional[List[float]] = None


That way we can also have some typing around it.

**Review by rbiseck3 — CHANGES_REQUESTED** [review]

File: unstructured/documents/elements.py [hunk 2]
```diff
@@ -454,9 +454,11 @@ def __init__(
         coordinate_system: Optional[CoordinateSystem] = None,
         metadata: Optional[ElementMetadata] = None,
         detection_origin: Optional[str] = None,
+        embeddings: Optional[List[float]] = None,
     ):
         metadata = metadata if metadata else ElementMetadata()
         self.text: str = text
+        self.embeddings: Optional[str] = embeddings
```
Comment (rbiseck3) [inline unstructured/documents/elements.py:461]: Why is the typing here set to `Optional[str]`?

Comment (ryannikolaidis) [reply unstructured/documents/elements.py:461]: Typo

Comment (ryannikolaidis) [reply unstructured/documents/elements.py:461]:
```suggestion
        self.embeddings: Optional[List[float]] = embeddings
```

**Review by rbiseck3 — APPROVED** [review]

**Pull request merged**